### PR TITLE
Fixes for issue #134 and fixing tests

### PR DESCRIPTION
**Pull request by wgoulet**
This PR attempts to address the issues reported in #134 . For this fix, I also included some minor updates to the docs to indicate all the environment variables that should be set when running 'make tpp_test'. I also added some new tests for the cmd package and added an option to the makefile to be able to run those tests with make. Finally, I fixed a broken test in the cmd package where the path separator was missing from a call to create a local file.

I am not setup to run cross-platform tests across all 3 platforms and the only test I am unable to execute successfully is Test_GetCertificateList in pkg/venafi/tpp due to not having a system with a large enough list of certificates in a single policy folder to test against. Other than these comments, all TPP tests pass on Linux as well as all cmd/vcert tests pass.

### CODE REVIEW

File: cmd/vcert/validators.go [hunk 1]
```diff
@@ -58,9 +59,15 @@ func validateCommonFlags(commandName string) error {
 		return fmt.Errorf("The '-file' option cannot be used used with any other -*-file flags. Either all data goes into one file or individual files must be specified using the appropriate flags")
 	}
 
-	csrOptionRegex := regexp.MustCompile(`^file:.*$|^local$|^service$|^$`)
+	csrOptionRegex := regexp.MustCompile(`(^file:).*$|^local$|^service$|^$`)
 	if !csrOptionRegex.MatchString(flags.csrOption) {
-		return fmt.Errorf("unexpected -csr option: %s", flags.csrOption)
+		return fmt.Errorf("unexpected -csr option provided: %s; specify one of the following options: %s, %s, or %s", flags.csrOption, "'file:<filename>'", "'local'", "'service'")
```
Comment (tr1ck3r) [inline cmd/vcert/validators.go:64]: Even though we still allow the single dash for backward compatibility, the double dash is the proper syntax so should be `--csr` here.

File: cmd/vcert/validators.go [hunk 2]
```diff
@@ -58,9 +59,15 @@ func validateCommonFlags(commandName string) error {
...
+	}
+
+	csrOptFlagResults := csrOptionRegex.FindStringSubmatch(flags.csrOption)
+
+	if csrOptFlagResults[1] != "" && (flags.keyTypeString != "" || flags.keyCurveString != "" || flags.keySize > 0) {
+		return fmt.Errorf("the '-keytype','-keycurve' and 'key-size' options cannot be used when '-csr file:' option is provided")
```
Comment (tr1ck3r) [inline cmd/vcert/validators.go:70]: Should be `--key-type`, `--key-curve`, `--key-size`, and `--csr`.

Comment (wgoulet) [reply cmd/vcert/validators.go]: @tr1ck3r Updated the error messages with the -- as requested.

**Review by tr1ck3r — APPROVED** [review]
Changes look good to me

**Review by rvelaVenafi — APPROVED** [review]
Very nice test you did here. Thanks a lot.

**Review by angelmoo — APPROVED** [review]
code looks good

**Review by rvelaVenafi — APPROVED** [review]
Changes look good. 
Thanks for your contribution!

**Comment by wgoulet** [comment]
@rvelaVenafi Thanks I screwed up in github UI and accidentally set you to re-review; pls ignore that.

**Comment by rvelaVenafi** [comment]
> @rvelaVenafi Thanks I screwed up in github UI and accidentally set you to re-review; pls ignore that.

No worries!

**Pull request merged**